### DISCUSSION TITLE
3D Gan question

### DISCUSSION THREAD

**Discussion by davidiommi** [Project-MONAI/MONAI#2022] [Q&A]
Hello,

I was reading a previous thread about the implementation of GANs in Monai. Does Monai support 3D GANs? Is a pix2pix or a dcGan implementation? 

The tutorial shows a 2D GAN, but I suppose it's enough to the change the setting for the 3D case

Thanks in advance

**Comment by Nic-Ma** [comment]
Hi @ericspod ,

I think the GAN networks in MONAI support both 2D and 3D.
Could you please help double confirm?

Thanks.

**Comment by ericspod** [comment]
The `Generator` and `Discriminator` networks do support 2D and 3D. They and other networks require arguments to state the input size and the dimensionality of the network is inferred from that. For `Generator` this is `start_shape`, which if it has 4 values for (channel, depth, width, height) values will produce a 3D network. For `Discriminator` and the other subtypes of `Regressor` this is `in_shape`.
